src/Hosting/Hosting/src/Internal/ErrorPageBuilder.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Hosting.Views;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.StackTrace.Sources;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    internal static class ErrorPageBuilder
+    {
+        public static RequestDelegate BuildErrorPageApplication(
+            IFileProvider contentRootFileProvider,
+            ILogger logger,
+            bool showDetailedErrors,
+            Exception exception)
+        {
+            if (exception is TargetInvocationException tae)
+            {
+                exception = tae.InnerException!;
+            }
+
+            var model = ErrorPageModelBuilder.CreateErrorPageModel(contentRootFileProvider, logger, showDetailedErrors, exception);
+
+            var errorPage = new ErrorPage(model);
+            return context =>
+            {
+                context.Response.StatusCode = 500;
+                context.Response.Headers[HeaderNames.CacheControl] = "no-cache,no-store";
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Preventing_caching recommends either a vanilla `no-store` or a `no-store, max-age=0`

Reply:
Don't change this as part of this PR, file an issue @pranavkm